### PR TITLE
Test fetch pool price

### PR DESCRIPTION
**Pull request by enriquefynn**
Add some tests for the `FetchPoolPrice` instruction.

### CODE REVIEW

File: anker/tests/tests/fetch_pool_price.rs [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    context
+        .solido_context
+        .advance_to_normal_epoch(2 + POOL_PRICE_NUM_SAMPLES as u64);
+    context.fetch_pool_price().await;
+    let anker = context.get_anker().await;
+    assert_eq!(
+        anker.historical_st_sol_prices.0[..],
+        expected_historical_st_sol_prices
+    );
+}
```
Comment (ruuda) [inline anker/tests/tests/fetch_pool_price.rs:54]: Can we also add a test that swaps against the pool to change the price, and check that we sample that new price as expected?

File: anker/tests/tests/fetch_pool_price.rs [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        .initialize_token_pool_and_deposit(Lamports(DEPOSIT_AMOUNT))
+        .await;
+    context.fetch_pool_price().await;
+    context
+        .solido_context
+        .context
+        .warp_to_slot(956256 + POOL_PRICE_MIN_SAMPLE_DISTANCE - 1)
+        .expect("Failed to warp to slot");
+    let result = context.try_fetch_pool_price().await;
+    assert_solido_error!(result, AnkerError::FetchPoolPriceTooEarly);
```
Comment (ruuda) [inline anker/tests/tests/fetch_pool_price.rs:69]: To confirm that the test works, we should also warp to slot `956256 + POOL_PRICE_MIN_SAMPLE_DISTANCE` and confirm that that one slot later, we _can_ fetch the price.

**Review by ruuda — COMMENTED** [review]
Thanks for adding tests! I have two suggestions for extending them.

**Comment by enriquefynn** [comment]
@ruuda: can you check with me why test `test_fetch_pool_price_when_price_changed` doesn't pass?
There's `9090.909091 UST` and `11 stSOL`, so the price for swapping `1 stSOL` should be more or less`8333.33333 UST`:
```
9090.909091 * 11 = 100000.000001
8333.33333 * 12 = 99999.999996
```
But in the test it's `757.575757`, but `757.575757 * 12 = 9090.909083999999`
I'm probably missing something on how this AMM works.

File: anker/tests/tests/fetch_pool_price.rs [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+    let mut context = Context::new().await;
+    context
+        .initialize_token_pool_and_deposit(Lamports(DEPOSIT_AMOUNT))
+        .await;
+    for epoch in 2..2 + POOL_PRICE_NUM_SAMPLES as u64 {
+        context.solido_context.advance_to_normal_epoch(epoch);
+        context.fetch_pool_price().await;
+    }
+    let anker = context.get_anker().await;
+    let current_ust_price = MicroUst(909090909);
```
Comment (ruuda) [inline anker/tests/tests/fetch_pool_price.rs:28]: Can you add a comment to explain where this number comes from?

**Review by ruuda — COMMENTED** [review]
The structure of the tests looks good, but I didn’t verify the numbers ... can you add a comment to explain how the initial stSOL price emerges? That would make it a bit easier for me to check.

**Review by ruuda — APPROVED** [review]
LGTM!

**Pull request merged**